activerecord/lib/active_record/associations/collection_proxy.rb

```diff
@@ -958,7 +958,7 @@ def scope
       #   other = [Pet.new(id: 1), Pet.new(id: 2)]
```

Comment:
I think the example still adds value, but maybe we should add a note.
What do you think?
```suggestion
      #
      # Note that unpersisted records can still be seen as equal:
      #
      #   other = [Pet.new(id: 1), Pet.new(id: 2)]
```